src/components/pagination/pagination.jsx

```diff
@@ -78,9 +76,12 @@ const PaginationButton = ( {
 } ) => {
 	const { size, disabled } = usePageContext();
 
+	// Set the tag to 'button' when disabled to avoid clickable links
+	const Tag = disabled ? 'button' : tag;
+
 	return (
 		<Button
-			tag={ tag }
+			tag={ Tag }
```

Comment:
No need of this casesensitive aproach, you can directly use like 
tag={ disabled ? 'button' : tag }